### ISSUE TITLE
Problems with Hist decStart restart tests due to variable orbital year

### ISSUE THREAD

**Issue by ekluzek** [ESCOMP/ctsm#379]
In clm5.0.dev010 I have problems with the following test:

ERP_D_P48x1.f10_f10_musgs.IHistClm50Bgc.hobart_nag.clm-decStart

I ran it three times. The first and last it failed the exact restart test, for a bunch of fields. So it is a problem. The second time it failed on the build of cism, which might have just been a random system glitch.

Since, I don't see other problems I'm going ahead with dev010, but we need to continue to investigate this.

**Comment by ekluzek** [comment]
OK, this is a bigger problem than I originally thought it was. It shows up for other compilers as well as cheyenne_intel. A PEM test shows only uvel and vvel changing, but a ERS test shows the problem. Changing decStart to default and the problem goes away. It also passes if the compset is changed from IHistClm50Bgc to I1850Clm50Bgc. So it's a likely a end or year problem for transient cases. It covers a large number of fields and looking at the map of the changes it's spread out all over the world.

I'm still going to move forward, but we really need to get to the bottom of this.

**Comment by billsacks** [comment]
@ekluzek From looking at your tests (if I'm looking in the right place), it looks like this very similar one passed, right? `ERP_D.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart`

I know you have a lot on your plate right now; let me know if you'd like some help working on this issue.

**Comment by billsacks** [comment]
This is suspicious: Looking at diffs in the cpl hist file in `/glade/p/work/erik/ctsm/cime/scripts/cases/ERS_D_P36x1.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart.GC.20180514_211330_kh78gx`, I see these fields (among others):

```
 RMS a2x_Faxa_swndr                   5.0120E-05            NORMALIZED  6.8136E-07
 RMS a2x_Faxa_swvdr                   4.0128E-05            NORMALIZED  6.6611E-07
 RMS a2x_Faxa_swndf                   2.5538E-05            NORMALIZED  1.6080E-06
 RMS a2x_Faxa_swvdf                   3.4720E-05            NORMALIZED  1.1891E-06
```

Of course, this is the cpl hist at the end of the run, but I can't see how other problems would affect datm, so I'd suspect these datm fields as the initial culprit.

**Comment by ekluzek** [comment]
@billsacks I think this is an important problem to get to the bottom of. If you could look at it that would be great. We should just try not to duplicate efforts. So update the issue with things you do/find.

I am pretty sure it's due to a very recent update in cime, so if there's something going haywire in datm that would mesh with that. That's part of why I figured we should move forward, since it's likely a cime issue.

**Comment by billsacks** [comment]
Okay, I'm starting to look into it.

**Comment by billsacks** [comment]
It looks like the problem is that datm (and maybe more... to be determined) don't restart properly with variable-year orbit. I'm working with @mvertens on this now.

**Comment by billsacks** [comment]
I have rerun `ERS_D_P36x1.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart` with a fix in datm (billsacks/cime@a67dceae9). The a2x fields no longer differ with this fix in place, but there are still differences between the base and restart test. So I'm thinking there may be a problem in CTSM, too. I'll look further.

**Comment by billsacks** [comment]
Okay, with my datm fix together with this change (which undoes the fix in #19 ):

```diff
diff --git a/src/main/clm_initializeMod.F90 b/src/main/clm_initializeMod.F90
index 393e2d0..443ddf3 100644
--- a/src/main/clm_initializeMod.F90
+++ b/src/main/clm_initializeMod.F90
@@ -378,7 +378,7 @@ contains
     ! given by the obliquity use negative value for S. Hem

     do g = bounds_proc%begg,bounds_proc%endg
-       max_decl = obliqr
+       max_decl = 0.409571
        if (grc%lat(g) < 0._r8) max_decl = -max_decl
        grc%max_dayl(g) = daylength(grc%lat(g), max_decl)
     end do
```

I am able to get these two tests to pass:

`ERS_D_P36x1.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart`

`ERP_D_P48x1.f10_f10_musgs.IHistClm50Bgc.hobart_nag.clm-decStart`

Now I need to put in a more robust fix for max_dayl (see #260 )

**Comment by ekluzek** [comment]
It looks like more tests were affected by this problem then I originally reported. Here are some additional tests that have this issue as well...

    FAIL ERP_D.f10_f10_musgs.IHistClm50Bgc.cheyenne_intel.clm-decStart COMPARE_base_rest
    FAIL ERP_Ly3_P72x2.f10_f10_musgs.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
    FAIL ERP_P36x2_D_Ld5.f10_f10_musgs.IHistClm45Bgc.cheyenne_intel.clm-decStart COMPARE_base_rest
    FAIL ERS_D_Ld7_Mmpi-serial.1x1_smallvilleIA.IHistClm50BgcCropGs.cheyenne_intel.clm-decStart1851_noinitial COMPARE_base_rest
    FAIL ERS_Ly3_Mmpi-serial.1x1_smallvilleIA.IHistClm50BgcCropGs.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
    FAIL ERS_Ly3_P72x2.f10_f10_musgs.IHistClm50BgcCropG.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
    FAIL ERS_Ly5_P144x1.f10_f10_musgs.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
    FAIL ERS_Ly5_P72x1.f10_f10_musgs.IHistClm45BgcCrop.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
    FAIL ERS_Ly6_Mmpi-serial.1x1_smallvilleIA.IHistClm50BgcCropGs.cheyenne_intel.clm-cropMonthOutput COMPARE_base_rest
